src/wakefs/fuse.cpp

```diff
@@ -217,16 +224,56 @@ bool run_in_fuse(fuse_args &args, int &status, std::string &result_json) {
   (void)close(STDOUT_FILENO);
   (void)close(STDERR_FILENO);
 
-  do waitpid(pid, &status, 0);
-  while (WIFSTOPPED(status));
+  bool has_timeout = args.command_timeout > 0;
+  pid_t timeout_pid = -1;
+
+  // Launch timer process
+  if (has_timeout) {
+    timeout_pid = fork();
+    if (timeout_pid < 0) {
+      std::cerr << "wakebox: failed to fork timout process" << std::endl;
+      exit(1);
+    }
+
+    if (timeout_pid == 0) {
+      execve_wrapper({"/usr/bin/sleep", std::to_string(args.command_timeout)}, {});
+    }
+  }
+
+  pid_t wait_pid;
+  while ((wait_pid = wait(&status)) != -1) {
+    if (wait_pid == timeout_pid && WIFEXITED(status)) {
+      kill(payload_pid, SIGKILL);
+      std::cerr << "wakebox: Timed out waiting for process to complete" << std::endl;
```

Comment:
I think we should avoid printing, as this will end up in `wake --failed` as text, but it's already being exposed as data in the result json which the `Runner` type in wake can handle